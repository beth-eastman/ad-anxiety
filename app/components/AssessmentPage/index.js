/* eslint-disable */
/**
 * AssessmentPage contains a list of assessments where each list item directs to
 * a specific page containing that assessment.
 *
 * Bethany Eastman
 */
import React from 'react';
import { GridList, GridTile } from 'material-ui';
import { assessments } from 'data/assessments';


const styles = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  gridList: {
    overflowY: 'auto',
  },
};

// Get number of columns needed for responsive grid list
const getCols = (width) => {
  if (!width) {
    return 1;
  } else if (width > 1280) {
    return 4;
  } else if (width > 900) {
    return 3;
  } else if (width > 600) {
    return 2;
  }
  return 1;
};

export default class AssessmentPage extends React.Component {

  /* Set number of columns */
  constructor(props) {
    super(props);

    this.state = {
      cols: getCols(window.innerWidth),
    };
  }

  /* Change AppBar title */
  componentWillMount() {
    this.props.setPageTitle("Assessments");
    this.props.changeIcon('Home');
  }

  componentDidMount() {
    // Add event listener for number of columns when window resizes
    const that = this;
    window.addEventListener(
      'resize',
      function () {
        setTimeout(
          function() {
            that.setState({ cols: getCols(window.innerWidth) });
          }
        , 250);
      }
    );
  }

  componentWillUnmount() {}

  /* Create a list for each assessment, where each list item contains a link
   * to the corresponding assessment page  */
  render() {
    return (
      <div style={styles.root}>
        <GridList
          style={styles.gridList}
          cols={this.state.cols}
        >
          {assessments.map((tile) => (
            <GridTile
              key={tile.title}
              title={tile.title}
              onTouchTap={() => {
                this.props.router.push('/assessment/'.concat(tile.id));
              }}
            >
              <img src={tile.img} alt={tile.title} />
            </GridTile>
          ))}
        </GridList>
      </div>
    );
  }
}
