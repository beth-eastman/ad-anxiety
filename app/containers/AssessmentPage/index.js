/**
 * Contains assessments for the application.
 */

import React from 'react';
import { GridList, GridTile } from 'material-ui';
import { browserHistory } from 'react-router';

// images // TODO: get correct images for assessment page
import UnderstandingAnxiety from 'components/img/anxiety-understanding-anxiety.jpg';
import AnxietyTreatment from 'components/img/anxiety-types-of-anxiety.jpg';
import TreatmentOptions from 'components/img/anxiety-treatment-options.jpg';
import RecognizingTrigers from 'components/img/anxiety-recognizing-triggers.jpg';
import BackInControl from 'components/img/anxiety-back-in-control.jpg';


const styles = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  gridList: {
    overflowY: 'auto',
  },
};

const assessments = [
  {
    id: 'anxiety',
    img: UnderstandingAnxiety,
    title: 'Anxiety',
  },
  {
    id: 'panic',
    img: AnxietyTreatment,
    title: 'Panic',
  },
  {
    id: 'worry',
    img: TreatmentOptions,
    title: 'Worry',
  },
  {
    id: 'depression',
    img: BackInControl,
    title: 'Depression',
  },
  {
    id: 'sleep',
    img: RecognizingTrigers,
    title: 'Sleep',
  },
  {
    id: 'pts',
    img: AnxietyTreatment,
    title: 'Post-Traumatic Stress',
  },
  {
    id: 'stress',
    img: TreatmentOptions,
    title: 'Stress',
  },
  {
    id: 'stigma',
    img: RecognizingTrigers,
    title: 'Stigma',
  },
];

export default class AssessmentPage extends React.Component {     //eslint-disable-line

  componentWillMount() {
    this.props.setPageTitle("Assessments"); // eslint-disable-line
  }

  render() {
    return (
      <div style={styles.root}>
        <GridList
          style={styles.gridList}
          cols={1}
        >
          {assessments.map((tile) => (
            <GridTile
              key={tile.title}
              title={tile.title}
              onTouchTap={() => {
                browserHistory.push('/assessment/'.concat(tile.id).concat('-assessment'));
              }}
            >
              <img src={tile.img} alt={tile.title} />
            </GridTile>
          ))}
        </GridList>
      </div>
    );
  }
}
