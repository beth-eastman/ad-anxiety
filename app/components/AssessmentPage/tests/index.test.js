/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import AssessmentPage from '../index';

it('renders AssessmentPage', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <AssessmentPage setPageTitle={function () {}} changeIcon={function () {}} />
    </MuiThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
