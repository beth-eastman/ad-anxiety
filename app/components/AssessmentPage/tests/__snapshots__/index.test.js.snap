exports[`test renders AssessmentPage 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "margin": -2,
        "muiPrepared": true,
        "overflowY": "auto",
      }
    }>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Anxiety"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Anxiety
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Panic"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Panic
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Worry"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Worry
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Depression"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Depression
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Sleep"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Sleep
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Post-Traumatic Stress"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Post-Traumatic Stress
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Stress"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Stress
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Stigma"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Stigma
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
